concordium-std/src/test_infrastructure.rs

```diff
@@ -530,3 +533,222 @@ pub fn report_error(message: &str, filename: &str, line: u32, column: u32) {
 #[doc(hidden)]
 #[cfg(not(all(feature = "wasm-test", target_arch = "wasm32")))]
 pub fn report_error(_message: &str, _filename: &str, _line: u32, _column: u32) {}
+
+/// Contract state for testing, mimicking the operations the scheduler allows,
+/// including the limit on the size of the maximum size of the contract state.
+pub struct ContractStateTest<T> {
+    pub cursor: Cursor<T>,
+}
+
+/// A borrowed instantiation of `ContractStateTest`.
+pub type ContractStateTestBorrowed<'a> = ContractStateTest<&'a mut Vec<u8>>;
+
+/// An owned variant that can be more convenient for testing since the type
+/// itself owns the data.
+pub type ContractStateTestOwned = ContractStateTest<Vec<u8>>;
+
+#[derive(Debug, PartialEq, Eq)]
+// An error that is raised when operating with `Seek`, `Write`, or `Read` trait
+// methods of the `ContractStateTest` type.
```

Comment:
Should probably be a doc comment
```suggestion
/// An error that is raised when operating with `Seek`, `Write`, or `Read` trait
/// methods of the `ContractStateTest` type.
```